### PR TITLE
docs: Clarify security boundaries in privileged mode 

### PR DESCRIPTION
**Pull request by leoluk**
See https://github.com/kata-containers/runtime/issues/1568

### CODE REVIEW

File: Limitations.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 * [Overview](#overview)
-* [Definiton of a limitation](#definiton-of-a-limitation)
+* [Definition of a limitation](#definiton-of-a-limitation)
```
Comment (grahamwhaley) [inline Limitations.md]: Nice pickup. Could you also fix the actual heading this points to (line 48) as well? :-)
Might be nice if you split this into a separate little commit as well - fine to roll it into this PR, just have two commits in this PR. thx!

File: Limitations.md [hunk 2]
```diff
@@ -221,11 +221,15 @@ See more documentation at
...
 Privileged support in Kata is essentially different from `runc` containers.
 Kata does support `docker run --privileged` command, but in this case full access
-to the guest VM is provided instead of the host.
+to the guest VM is provided in addition to some host access.
+
 The container runs with elevated capabilities within the guest and is granted 
 access to guest devices instead of the host devices.
 This is also true with using `securityContext privileged=true` with Kubernetes.
 
+The container may also be granted full access to all host devices
```
Comment (grahamwhaley) [inline Limitations.md]: Need to check with @amshinde if this is fully true, that it gets access to **all** host devices or not...

Comment (leoluk) [reply Limitations.md]: It does in fact mount all host devices into the VM (as long as they _can_ be mounted).

**Comment by leoluk** [comment]
Sure, updated the PR.

**Comment by leoluk** [comment]
Some feedback: That was a lot of work for a minor drive-by documentation change. I had to find and read the source code to understand this error message and get past the CI: 

https://github.com/kata-containers/tests/blob/master/cmd/checkcommits/checkcommits.go#L251-L253

**Comment by grahamwhaley** [comment]
/test
Thanks @leoluk - I know it's a high entry barrier for tiny changes, but it has served the project well in maintaining the quality of review and therefore code, and the CI has saved our necks a number of times :-)
If you have any suggestion (or PRs ;-) ) to improve the docs or that piece of code, feel free to drop them by...

**Comment by leoluk** [comment]
Not disagreeing, just pointing it out :)

PR good to merge?

**Pull request merged**